DuckDuckGo/AppDelegate.swift

```diff
@@ -33,6 +33,13 @@ import Core
 
     var window: UIWindow?
 
+    override init() {
+        super.init()
+        NotificationCenter.default.addObserver(forName: .databaseDidEncounterInsufficientDiskSpace, object: nil, queue: .main) { _ in
+            self.application(UIApplication.shared, willTerminateWithReason: .insufficientDiskSpace)
```

Comment:
AppDelegate is retained here. It should be present throughout app lifecycle regardless but I'd probably advise to not keep it strongly via closure.